### PR TITLE
Assign id subcommand

### LINKED ISSUE #30: Add new assign-id subcommand
This command would be responsible for assigning RUSTSEC ids to new vulnerabilities, which is currently done by hand (at least, as far as I can tell @tarcieri?).

It would probably work like this:

1. Find the current highest assigned id number for the current calendar year
2. Find vulnerability reports with an ID of `RUSTSEC-0000-0000`
3. Assign each of them an ID by a) renaming the file, b) rewriting the `id = ` line of the file

### PR DESCRIPTION
**Pull request by RestitutorOrbis**
A PR to resolve #30 you can use it by writing the following command `rustsec-admin assign-id <insert repo path here>`. The files it fills are files whose filename and ID are placeholder RUSTSEC IDs.

### CODE REVIEW

**Comment by RestitutorOrbis** [comment]
@alex Do you have time to take a look at this to see if it matches your intent with #30?

**Comment by alex** [comment]
Sorry, didn't get a notification on this for some reason. This looks like what I was expecting!

cc: @tarcieri

File: src/assigner.rs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+fn extract_id_num_from_str(id: Id) -> Option<u64> {
+    if let Kind::RUSTSEC = id.kind() {
+        let id_str = id.to_string();
+        let len = id_str.len();
+        let numerical_portion = &id_str[(len - 4)..len];
+        Some(numerical_portion.parse().unwrap())
+    } else {
+        None
+    }
+}
```
Comment (tarcieri) [inline src/assigner.rs:81]: It'd be good to add an upstream API for this to the `rustsec` crate. Not saying that's a blocker for this PR, but I was surprised there wasn't one.

Comment (RestitutorOrbis) [reply src/assigner.rs:81]: I've filed an issue here https://github.com/RustSec/rustsec-crate/issues/184#issue-638358836

Comment (tarcieri) [reply src/assigner.rs:81]: I've just released `rustsec` v0.20.1 which adds an API for this: https://docs.rs/rustsec/0.20.1/rustsec/advisory/id/struct.Id.html#method.numerical_part

Mind updating the PR to use it?

**Comment by tarcieri** [comment]
Apologies for not more thoroughly reviewing this yet. I hope to review it tomorrow!

**Comment by RestitutorOrbis** [comment]
> Apologies for not more thoroughly reviewing this yet. I hope to review it tomorrow!

No worries!

**Comment by tarcieri** [comment]
@RestitutorOrbis thank you!

**Comment by alex** [comment]
@tarcieri let me know when you tag a release, I'll write the github action to integrate this over in the advisory DB.

**Comment by tarcieri** [comment]
Might try to incorporate https://github.com/RustSec/rustsec-crate/pull/189 and cut a `rustsec` crate release first, but hopefully today!

**Pull request merged**